app/uk/gov/hmrc/alcoholdutyreturns/models/returns/AdrReturnSubmission.scala

```diff
@@ -247,7 +427,56 @@ case class AdrTotals(
   drawback: BigDecimal,
   repackagedDraught: BigDecimal,
   totalDutyDue: BigDecimal
-)
+) {
+  // Could do some additional checks here that values are 0/non-zero when declared in the appropriate section
+  def validate(
+    wasDutyDeclared: Boolean,
+    wasOverDeclaration: Boolean,
+    wasUnderDeclaration: Boolean,
+    wasSpoiltDeclaration: Boolean,
+    wasDrawbackDeclaration: Boolean,
+    wasRepackagedDraughtDeclaration: Boolean
+  ): ValidationResult = {
+    val wasNilReturn =
+      !(wasDutyDeclared || wasOverDeclaration || wasUnderDeclaration || wasSpoiltDeclaration || wasDrawbackDeclaration || wasRepackagedDraughtDeclaration)
+
+    (if (!wasDutyDeclared) {
+       Validations.validateZero(declaredDutyDue, "dutyDeclared.declaredDutyDue")
+     } else {
+       Validations.validateNonNegative(declaredDutyDue, "dutyDeclared.declaredDutyDue")
+     }) |+|
+      (if (!wasOverDeclaration) {
+         Validations.validateZero(overDeclaration, "dutyDeclared.overDeclaration")
+       } else {
+         Validations.validateNonPositive(overDeclaration, "dutyDeclared.overDeclaration")
+       }) |+|
+      (if (!wasUnderDeclaration) {
+         Validations.validateZero(underDeclaration, "dutyDeclared.underDeclaration")
+       } else {
+         Validations.validateNonNegative(underDeclaration, "dutyDeclared.underDeclaration")
+       }) |+|
+      (if (!wasSpoiltDeclaration) {
+         Validations.validateZero(spoiltProduct, "dutyDeclared.spoiltProduct")
+       } else {
+         Validations.validateNonPositive(spoiltProduct, "dutyDeclared.spoiltProduct")
+       }) |+|
+      (if (!wasDrawbackDeclaration) {
+         Validations.validateZero(drawback, "dutyDeclared.drawback")
+       } else {
+         Validations.validateNonPositive(drawback, "dutyDeclared.drawback")
+       }) |+|
+      (if (!wasRepackagedDraughtDeclaration) {
+         Validations.validateZero(repackagedDraught, "dutyDeclared.repackagedDraught")
+       } else {
+         noValidationResult
+       }) |+|
+      (if (wasNilReturn) {
+         Validations.validateZero(totalDutyDue, "dutyDeclared.totalDutyDue")
+       } else {
+         noValidationResult
```

Comment:
This is the grand total. If you have more adjustment claims in your favour than things owing (duty + adjustments not in your favour) it can be negative. Only time we know something about it is a nil return.